resources/js/components/tools/csv/CsvSettings.vue

```diff
@@ -0,0 +1,283 @@
+                required: false,
+                default: 10,
+            },
+            skippedCount: {
+                type: Number,
+                required: false,
+                default: 0,
+            },
+            maxRows: {
+                type: Number,
+                required: false,
+                default: 100,
+            },
+            showPreview: {
+                type: Boolean,
+                required: false,
+                default: true,
+            },
+            total: {
+                type: Number,
+                required: true
+            }
+        },
+        emits: [
+            'update:delimiter',
+            'update:hasHeaderRow',
+            'update:showLinenumbers',
+            'update:useCustomDelimiter',
+            'update:showCount',
+            'update:skippedCount',
+            'update:showPreview',
+        ],
+        setup(props, context) {
+            const { t } = useI18n();
+
+
+            const delimiterOptions = [
+                { value: ',', label: ',' },
+                { value: ';', label: ';' },
+                { value: '\t', label: t('keyboard.tab') },
+                { value: ' ', label: t('keyboard.space') },
+                // { value: 'CUSTOM', label: 'Custom' }
+            ];
+
+            function changeCustomDelimiter() {
+                if(props.useCustomDelimiter) {
+                    const delimiterExists = delimiterOptions.find(option => option.value === props.delimiter);
+                    if(!delimiterExists)
+                        context.emit('update:delimiter', delimiterOptions[0].value);
+                }
+
+                context.emit('update:useCustomDelimiter', !props.useCustomDelimiter);
+            }
+
+            const delimiterInputStyle = {
+                width: '5em',
+                minWidth: '5em',
+                flex: '1 1 min-content',
+                zIndex: 2048,
+            };
+
+            const rangeInformationText = computed(() => {
+
+                const start = props.skippedCount + 1;
+                let end = props.skippedCount + props.showCount;
+                if(end > props.total) {
+                    end = props.total;
+                }
+
+                return `${start}-${end} ${t('global.paginator.of')} ${props.total}`;
+
+            });
+
+            const paginatorInputStyle = {
+                minWidth: '5em',
+            };
+
+
+            return {
+                t,
+                changeCustomDelimiter,
+                delimiterOptions,
+                delimiterInputStyle,
+                multiselectResetClasslist,
+                rangeInformationText,
+                paginatorInputStyle,
+            };
+        },
+
+    };
+</script>
+
+<style scoped>
+    .line {
+        display: flex;
+        flex: 1;
+        justify-content: flex-start;
+        align-items: center;
+        flex-wrap: nowrap;
+        gap: 2rem !important;
+    }
+</style>
```

Comment:
I think most of this can be done using bs classes

Reply:
k